cql3/statements/select_statement.cc

```diff
@@ -2258,7 +2343,11 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
                 prepare_limit(db, ctx, _per_partition_limit),
                 stats,
                 std::move(prepared_attrs));
-    } else if (restrictions->uses_secondary_indexing() || prepared_ann_ordering) {
+    } else if (prepared_ann_ordering) {
+        stmt = vector_indexed_table_select_statement::prepare(db, schema, ctx.bound_variables_size(), _parameters, std::move(selection), std::move(restrictions),
+                std::move(group_by_cell_indices), is_reversed_, std::move(ordering_comparator), std::move(*prepared_ann_ordering),
+                prepare_limit(db, ctx, _limit), prepare_limit(db, ctx, _per_partition_limit), stats, std::move(prepared_attrs));
+    } else if (restrictions->uses_secondary_indexing()) {
```

Comment:
If a table has both an ordinary secondary index on one column and a vector index on a second column, I think this code will work correctly to allow one SELECT using the ordinary index, and a different SELECT using the vector index. What happens if you try to use both in one select - do you get a reasonable behavior? E.g., will it use the vector search and then filter on the other column (without using the secondary index)?
If we don't already do, it would be nice to have a test for these cases of a single table with two kinds of index - to see it behave reasonably.